### PR TITLE
drawin: lazily configure drawin windows

### PR DESCRIPTION
**Pull request by blueyed**
Ref: https://github.com/awesomeWM/awesome/issues/411.
Source: https://github.com/awesomeWM/awesome/pull/174#commitcomment-12708326.

### CODE REVIEW

**Comment by psychon** [comment]
As I already wrote in the comment where I proposed this:

    (Now that I just read through the diff again, I think that drawin_moveresize() is wrong:
     It should change w->geometry first and only afterwards emit signals, so that the other
     properties are already updated when the signals occur, similar to how this is done in
     client_resize_do().)

If you want, I will write that later. I'm not on my computer right now.

Basically, I would do `area_t old_geometry = w->geometry` at the beginning of `drawin_moveresize` and at the end (after everything else that is already done) it should do e.g. `if (old_geometry.x != w->geometry.x) emit_property::x;`.

Otherwise e.g. changing the bounding or the clip shape inside a `property::x` signal won't work correctly. Sorry for not fixing this before I wrote that comment.

**Comment by blueyed** [comment]
It wasn't clear to me if the comment was meant for your patch or the diff from the other PR/commit.

> If you want, I will write that later. I'm not on my computer right now.

Please do.

**Comment by psychon** [comment]
Merged in 9408b61a974bc351e285592f36b2e325c66d5cf2 (with my changes)
